tracer/test/Datadog.Trace.Security.Unit.Tests/WafConcurrencyTests.cs

```diff
@@ -1,34 +1,74 @@
-// <copyright file="ContextTests.cs" company="Datadog">
+// <copyright file="WafConcurrencyTests.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
 using System;
 using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Threading;
+
 #if NETFRAMEWORK
 using System.Web.Routing;
-#else
 #endif
 using Datadog.Trace.AppSec;
-using Datadog.Trace.AppSec.Rcm;
 using Datadog.Trace.AppSec.Rcm.Models.AsmDd;
 using Datadog.Trace.AppSec.Waf;
 using Datadog.Trace.AppSec.Waf.Initialization;
+using Datadog.Trace.AppSec.Waf.ReturnTypes.Managed;
 using Datadog.Trace.Security.Unit.Tests.Utils;
 using Datadog.Trace.TestHelpers.FluentAssertionsExtensions.Json;
 using FluentAssertions;
-using Moq;
 using Xunit;
 
 namespace Datadog.Trace.Security.Unit.Tests;
 
-public class ContextTests : WafLibraryRequiredTest
+public class WafConcurrencyTests : WafLibraryRequiredTest
 {
     // here we use just 1 sec instead of the 20sec common one as we dont really care about the result, just that it runs
     public const int WafRunTimeoutMicroSeconds = 1_000_000;
 
+    [Fact]
+    public void MultipleWafOps()
+    {
```

Comment:
Just to confirm, _Without_ your fix in `Waf`, this reproduces the issue?

Reply:
Yes it does. The only issue is it crashes the whole testhost.exe which stops responding and we don't have clear visibility that it was an unmanaged crash, so if there's a (unmanaged) regression with it, instead of showing test fail with a failed assert the process is just going to stop responding. But I'm guessing it still means the test is failing 😅